sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientLiveTests.cs

```diff
@@ -678,44 +682,92 @@ public async Task CanQueryWithFormattedQuery(FormattableStringWrapper query)
 
-            bool verifyRow = false;
-            bool verifyColumn1 = false;
-            bool verifyColumn2 = false;
-            foreach (LogsTableRow rows in resultTable.Rows)
-            {
-                foreach (var row in rows)
-                {
-                    if ((row != null) && row.ToString().Contains("Create/Update Storage Account"))
-                    {
-                        verifyRow = true;
-                    }
-                }
-            }
+            // Make sure there is some data in the range specified
+            results = await client.QueryResourceAsync<DateTimeOffset>(
+                new ResourceIdentifier(TestEnvironment.WorkspacePrimaryResourceId),
+                $"{_logsTestData.TableAName} | distinct * | project {LogsTestData.TimeGeneratedColumnName}",
+                timespan);
 
-            foreach (LogsTableColumn columns in resultTable.Columns)
-            {
-                if (columns.Name == "SubscriptionId" && columns.Type == LogsColumnType.StringTypeValue)
-                    verifyColumn1 = true;
+            Assert.GreaterOrEqual(results.Value.Count, 3);
+        }
 
-                if (columns.Name == "TimeGenerated" && columns.Type == LogsColumnType.DatetimeTypeValue)
-                    verifyColumn2 = true;
-            }
+        [LiveOnly]
+        [Test]
+        public async Task CanQueryResourcePrimaryWorkspace()
+        {
+            var timespan = TimeSpan.FromSeconds(5);
+
+            var client = CreateClient();
+            // Empty check
+            Response<LogsQueryResult> results = await client.QueryResourceAsync(
+                new ResourceIdentifier(TestEnvironment.WorkspacePrimaryResourceId),
+                $"{_logsTestData.TableAName} | distinct * | project {LogsTestData.TimeGeneratedColumnName}",
+                timespan);
+
+            Assert.AreEqual(0, results.Value.Table.Rows.Count);
+
+            // Check if all rows in table were uploaded
+            // Get the time of the third event and add a bit of buffer to it (events are 2d apart)
+            var maxOffset = (DateTimeOffset)_logsTestData.TableA[2][LogsTestData.TimeGeneratedColumnNameSent];
+            timespan = Recording.UtcNow - maxOffset;
+            timespan = timespan.Add(TimeSpan.FromDays(7));
+
+            // Make sure there is some data in the range specified
+            results = await client.QueryResourceAsync(
+                new ResourceIdentifier(TestEnvironment.WorkspacePrimaryResourceId),
+                $"{_logsTestData.TableAName} | distinct * | project {LogsTestData.TimeGeneratedColumnName}",
+                timespan);
+
+            Assert.GreaterOrEqual(results.Value.Table.Rows.Count, 3);
+        }
 
-            Assert.IsTrue(verifyRow);
-            Assert.IsTrue(verifyColumn1 && verifyColumn2);
+        [LiveOnly]
+        [Test]
+        public async Task CanQueryResourceSecondaryWorkspace()
```

Comment:
Curious what is the difference between the primary and secondary workspace?

Do we also need a test for Secondary Workspace with the generic method?

Reply:
It is another workspace created for testing, I do have test methods for it